harness/determined/common/schemas/expconf/_v0.py

```diff
@@ -221,58 +221,84 @@ def __init__(
 DataLayerConfigV0.finalize(DataLayerConfigV0_Type)
 
 
-class EnvironmentImageV0(schemas.SchemaBase):
+class CudaCompatMixin:
+    @no_type_check
+    def _ensure_cuda_compat(self) -> None:
+        # Backwards compatibility.
+        if self.cuda is None and self.gpu is not None:
+            self.cuda = self.gpu
+        # Equalize values.
+        self.gpu = self.cuda
+
+
+class EnvironmentImageV0(schemas.SchemaBase, CudaCompatMixin):
     _id = "http://determined.ai/schemas/expconf/v0/environment-image.json"
     cpu: Optional[str] = None
+    cuda: Optional[str] = None
     gpu: Optional[str] = None
```

Comment:
I don't think that there should be a `gpu` field here anymore.

Instead, do the translation in `from_dict` instead, the same as the go code would do the translation while parsing.

Probably this means you can make your `_ensure_cuda_compat` into a simple function rather than a mixin.